src/nspawn/nspawn-settings.c

```diff
@@ -1019,3 +1021,57 @@ int config_parse_bind_user(
 
         return 0;
 }
+
+int parse_bind_user_shell(const char *s, char **ret) {
+        int r;
+        char *sh;
+
+        if (is_path(s))
+                sh = strdup(s);
+        else {
+                r = parse_boolean(s);
+                if (r < 0)
+                        return r;
+
+                sh = strdup(r ? "yes" : "no");
+        }
+        if (!sh)
+                return -ENOMEM;
+
+        *ret = sh;
+
+        return 0;
+}
+
+int config_parse_bind_user_shell(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Settings *settings = ASSERT_PTR(data);
+        int r;
+
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                settings->bind_user_shell = NULL;
+                return 0;
+        }
+
+        r = parse_bind_user_shell(rvalue, &settings->bind_user_shell);
```

Comment:
Also here. Previous assignment is leaked. Please use a temporal variable and `free_and_replace()` on success.

Reply:
:pray: Addressed in f998d33e456ca0e7448fe295a37195c6b79e7636.